File: wasm_tests/test_1.js
export function test_1() {
    function fibonacci(num) {
        if (num === 1) {
            return 1;
        } else if (num === 2) {
            return 1;
        } else return fibonacci(num - 1) + fibonacci(num - 2);
    }

    fibonacci(35);
}

File: wasm_tests/test_1.rs
pub fn test_1() {
    fn fibonacci(n: f64) -> f64 {
        match n {
            0_f64 => 1_f64,
            1_f64 => 1_f64,
            _ => fibonacci(n - 1_f64) + fibonacci(n - 2_f64),
        }
    }

    std::hint::black_box(fibonacci(35_f64));
}
